### PR TITLE
[Combobox] Fix presetting option

### LINKED ISSUE #4250: fix: Preset option in Combobox does not work/shows blank combobox
# 🐛 Bug Report

The initial value passed to the Combobox is not shown. The Combobox is empty on first render.

## 💻 Repro or Code Sample

On the demo website: https://www.fluentui-blazor.net/combobox
Pre-selected option example is empty.

## 🤔 Expected Behavior

The initial value of the Combobox should be shown.

## 😯 Current Behavior

The initial value is not shown.

## 🌍 Your Environment

* OS & Device: Windows on PC
* Browser: Microsoft Edge
* .NET and Fluent UI Blazor library Version: 9 and 4.13.0

### PR DESCRIPTION
**Pull request by vnbaaij**
Fix #4250 by extending the `GetOptionSelected` code

### CODE REVIEW

File: src/Core/Components/List/ListComponentBase.razor.cs [hunk 1]
```diff
@@ -483,14 +483,18 @@ protected virtual bool GetOptionSelected(TOption item)
...
-            else if (SelectedOption == null)
+            else if (SelectedOption == null && string.IsNullOrEmpty(Value))
             {
                 return false;
             }
             else if (OptionValue != null && SelectedOption != null)
             {
                 return GetOptionValue(item) == GetOptionValue(SelectedOption);
             }
+            else if (!string.IsNullOrEmpty(Value) || !string.IsNullOrEmpty(InternalValue))
```
Comment (Tyme-Bleyaert) [inline src/Core/Components/List/ListComponentBase.razor.cs]: Whats the use case of the extra null check : `!string.IsNullOrEmpty(InternalValue)`?
Or were you gonna do somthing like this: `return string.Equals(GetOptionValue(item), Value ?? InternalValue, StringComparison.Ordinal);`

Comment (vnbaaij) [reply src/Core/Components/List/ListComponentBase.razor.cs]: Oops...that was a leftover of trying to solve the other issue I mentioned. Will remove it

**Comment by dvoituron** [comment]
This `ListComponentBase` class is used by many other components (including Autocomplete). Can you confirm that this will not impact other components? In version 5, we need to add additional unit tests with all these edge cases (null, empty, etc.).

We should also add a unit test in the current version 4 to verify this right away.

**Comment by vnbaaij** [comment]
> This `ListComponentBase` class is used by many other components (including Autocomplete). Can you confirm that this will not impact other components? In version 5, we need to add additional unit tests with all these edge cases (null, empty, etc.).
> 

This change only touches the `GetOptionSelected` method and returns true in case the `Value` is equal to the `OptionValue` (invoked on the item)

**Comment by dvoituron** [comment]
> This change only touches the `GetOptionSelected` method and returns true in case the `Value` is equal to the `OptionValue` (invoked on the item)

That's not the only change. You also have a test on `IsNullOrEmpty(Value)`. This could impact other components or cases. 

How can we be sure there won't be any regressions?

**Comment by vnbaaij** [comment]
Value was not being considered in this method before this change. I don't see how that could lead to a regression.

**Comment by vnbaaij** [comment]
@microsoft-github-policy-service rerun

**Comment by vnbaaij** [comment]
@microsoft-github-policy-service agree
